File: src/staf/docs/education/handson/staf/solutions/STAFTestcase3.java
import com.ibm.staf.*;
public class STAFTestcase3
{
    public static void main(String[] args)
    {
        int counter = 10;  
        STAFHandle handle = null;
        try
        {
            handle = new STAFHandle("STAFTestcase");
        }
        catch(STAFException e)
        {
            e.printStackTrace();  
            System.exit(1);
        }
        if (args.length > 0) counter = (new Integer(args[0])).intValue();
        for (int i=0; i < counter; i++)
        {
            System.out.println("Loop #" + i);
            STAFResult result = handle.submit2("local", "monitor", 
                "log message " + 
                STAFUtil.wrapData("Loop #" + i));
            STAFResult logResult = handle.submit2("local", "log", 
                "log machine logname STAFTestcase1.log level info message " + 
                STAFUtil.wrapData("Loop #" + i));
            STAFResult delayResult = handle.submit2("local", "delay", 
                "delay 1000");
            STAFResult varResult = handle.submit2("local", "var", 
                "resolve {HandsOn/Variables/MyVar}");
            if ((varResult.rc == 0) && (varResult.result.equals("STAFTestcase3/Terminate")))
            {
                System.exit(0);
            }
        }
        System.exit(0);
    }
}

File: src/staf/docs/education/solutions/Part1/exercise1.8/STAFTestcase3.py
from PySTAF import *
import sys
import time

counter = 10

try:
    handle = STAFHandle("STAFTestcase")
except STAFException, e:
    print "Error registering with STAF, RC: %d" % e.rc
    sys.exit(e.rc)

print "Using handle " + str(handle.handle);

if (len(sys.argv) > 1):
    counter = sys.argv[1]

for i in range(int(counter)):
    print "Loop #" + str(i)
    result = handle.submit("local", "monitor",
        "log message " + wrapData("Loop #" + str(i)))
    result = handle.submit("local", "log",
        "log machine logname STAFTestcase1.log level info message " +
        wrapData("Loop #" + str(i)))
    result = handle.submit("local", "delay", "delay 1000")
    result = handle.submit("local", "var",
        "resolve string {HandsOn/Variables/MyVar}")
    if (result.rc == 0 and result.result == "STAFTestcase3/Terminate"):
        sys.exit(0)

sys.exit(0)
